include/nanobind/nb_misc.h

```diff
@@ -39,4 +39,12 @@ inline void set_implicit_cast_warnings(bool value) noexcept {
     detail::set_implicit_cast_warnings(value);
 }
 
+/// \ingroup python_builtins
+/// Return a dictionary representing the global variables in the current execution frame,
+/// or ``__main__.__dict__`` if there is no frame.
+inline dict globals() {
+    PyObject *p = PyEval_GetGlobals();
```

Comment:
isn't a simple: `return borrow<dict>(PyEval_GetGlobals());` enough? My understanding is that this will always work when nanobind code runs as an extension module, which is the case in nanobind projects.

Reply:
No idea. `PyEval_GetGlobals` _can_ return `NULL`, so I guess we should handle this case somehow? Perhaps throwing an error would be more appropriate.

Reply:
Yes, the way to handle it in that case is via ``nb::detail::raise_python_error()``.